src/datasets/utils/file_utils.py

```diff
@@ -153,7 +153,7 @@ def head_hf_s3(
             max_retries=max_retries,
         )
     except Exception as e:
-        return e
+        return e  # TODO: return or raise?
```

Comment:
@lhoestq is this a bug? The exception is returned instead of raised.

Reply:
No this is on purpose, we don't want any error to be raised from it in case there's no internet

Reply:
Then, this makes no sense: https://github.com/huggingface/datasets/blob/master/src/datasets/load.py#L431-L434

Reply:
Oh right !

Reply:
Let's try to change it to raise then to remove the confusion

Reply:
Now `head_hf_s3` may raise an exception. I fix `prepare_module` to ignore the exception when it is not pertinent.